File: api-tester/src/main/java/com/revenuecat/apitester/java/OfferPaymentModeAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.models.OfferPaymentMode;

@SuppressWarnings({"unused", "SpellCheckingInspection"})
final class OfferPaymentModeAPI {
    static void check(final OfferPaymentMode offerPaymentMode) {
        switch (offerPaymentMode) {
            case PAY_AS_YOU_GO:
            case PAY_UP_FRONT:
            case FREE_TRIAL:
        }
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/OfferPaymentModeAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.models.OfferPaymentMode

@Suppress("unused", "UNUSED_VARIABLE")
private class OfferPaymentModeAPI {
    fun check(offerPaymentMode: OfferPaymentMode) {
        when (offerPaymentMode) {
            OfferPaymentMode.DISCOUNTED_RECURRING_PAYMENT,
            OfferPaymentMode.SINGLE_PAYMENT,
            OfferPaymentMode.FREE_TRIAL,
            -> {
            }
        }.exhaustive
    }
}
